### PR TITLE
New Package Wonton

### PR DESCRIPTION
**Pull request by raovgarimella**

### CODE REVIEW

File: var/spack/repos/builtin/packages/wonton/package.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    depends_on('cuda', when='+cuda')
+
+    # Kokkos with appropriate option
+    depends_on('kokkos +openmp', when='+kokkos +openmp')
+    depends_on('kokkos +cuda', when='+kokkos +cuda')
+
+    def cmake_args(self):
+        options = []
+        if '+mpi' in self.spec:
+            options.append('-DWONTON_ENABLE_MPI=ON')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/wonton/package.py:68]: Can this be explicitly disabled in an else clause? Same for other variants. CMake likes to toggle defaults based on what is found on the OS, making builds non-deterministic and unrepeatable.

Comment (raovgarimella) [reply var/spack/repos/builtin/packages/wonton/package.py:68]: Done

**Review by adamjstewart — APPROVED** [review]

**Comment by raovgarimella** [comment]
Do I need to worry about the macOS tests failing?

**Comment by adamjstewart** [comment]
> Do I need to worry about the macOS tests failing?

Nope, we've been having issues with the macOS tests for a couple weeks now, still trying to track that down and fix it in another PR.

**Pull request merged**